packages/@ama-sdk/schematics/schematics/typescript/core/openapi-codegen-typescript/src/main/resources/typescriptFetch/api/api.mustache

```diff
@@ -74,7 +74,7 @@ export class {{classname}} implements Api {
     const getParams = this.client.extractQueryParams<{{#uppercaseFirst}}{{nickname}}{{/uppercaseFirst}}RequestData>(data, [{{#trimComma}}{{#queryParams}}'{{baseName}}', {{/queryParams}}{{/trimComma}}]{{^queryParams}} as never[]{{/queryParams}});
     const metadataHeaderAccept = metadata?.headerAccept || '{{#headerJsonMimeType}}{{#produces}}{{mediaType}}{{^-last}}, {{/-last}}{{/produces}}{{/headerJsonMimeType}}';
     const headers: { [key: string]: string | undefined } = { {{#trimComma}}
-      'Content-Type': metadata?.headerContentType || '{{#headerJsonMimeType}}{{#consumes}}{{mediaType}}, {{/consumes}}application/json{{/headerJsonMimeType}}',
+      'Content-Type': metadata?.headerContentType || '{{#headerJsonMimeType}}{{#consumes}}{{mediaType}}{{^-last}}, {{/-last}}{{/consumes}}{{/headerJsonMimeType}}',
```

Comment:
I guess that the previous purpose of `application/json` is to provide a default value in case `consumes` array is empty.
Your change suggest that the `consumes` is always populated with a default `application/json` if not specified in the spec, is it the case?
If not, an additiomal fallback mechanism to `application/json` may need to be added

Reply:
The ```headerJsonMimeType``` lambda returns only one mime, and it prioritizes the first json mime found.
In the regression, we give as inputs to this lambda an array composed of [...theMimesTakenFromTheSpecConsume, 'application/json']. So the lambda will always return a json mime, whatever the mimes defined in the spec are.

As you guessed, the above proposed change only removes the invalid concatenation and does not provide a fallback mechanism. For the fallback, we could update the HeaderJsonMimeType implementation to return the `application/json` mime if the input array is empty.